buttercup/cogs/stats.py

```diff
@@ -42,6 +48,121 @@ async def _stats(self, ctx: SlashContext) -> None:
+                "Defaults to 24 hours.",
+                option_type=4,
+                required=False,
+            ),
+        ],
+    )
+    async def _progress(
+        self, ctx: SlashContext, username: Optional[str] = None, hours: int = 24,
+    ) -> None:
+        """Get the transcribing progress of a user in the given time frame."""
+        start = datetime.now()
+        user = username or extract_username(ctx.author.display_name)
+        # Send a first message to show that the bot is responsive.
+        # We will edit this message later with the actual content.
+        msg = await ctx.send(
+            i18n["progress"]["getting_progress"].format(user=user, hours=hours)
+        )
+
+        volunteer_response = self.blossom_api.get_user(user)
+        if volunteer_response.status != BlossomStatus.ok:
+            await msg.edit(content=i18n["progress"]["user_not_found"].format(user))
+            return
+        volunteer_id = volunteer_response.data["id"]
+
+        if volunteer_response.data["gamma"] == 0:
+            # The user has not started transcribing yet
+            await msg.edit(
+                content=i18n["progress"]["embed_message"].format(
+                    get_duration_str(start)
+                ),
+                embed=Embed(
+                    title=i18n["progress"]["embed_title"].format(user),
+                    description=i18n["progress"]["embed_description_new"].format(
+                        user=user
+                    ),
+                ),
+            )
+            return
+
+        one_day_ago = start - timedelta(hours=hours)
+
+        # We ask for submission completed by the user in the last 24 hours
+        # The response will contain a count, so we just need 1 result
+        progress_response = self.blossom_api.get(
+            "submission/",
+            params={
+                "completed_by": volunteer_id,
+                "from": one_day_ago.isoformat(),
+                "page_size": 1,
+            },
+        )
+        if progress_response.status_code != 200:
+            await msg.edit(
+                content=i18n["progress"]["failed_getting_progress"].format(user)
+            )
+            return
+        progress_count = progress_response.json()["count"]
+
+        if hours != 24:
+            # If it isn't 24, we can't really display a progress bar
+            await msg.edit(
+                content=i18n["progress"]["embed_message"].format(
+                    get_duration_str(start)
+                ),
+                embed=Embed(
+                    title=i18n["progress"]["embed_title"].format(user),
+                    description=i18n["progress"]["embed_description_other"].format(
+                        count=progress_count, hours=hours,
+                    ),
+                ),
+            )
+            return
+
+        motivational_messages = i18n["progress"]["motivational_messages"]

```

Comment:
I think the retrieval of the motivational message should be moved to a separate function

Reply:
Good point